projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -68,24 +70,32 @@ class ProjectionRegistryActor extends Actor with ActorLogging {
   val writeConsistency: WriteConsistency = WriteMajority(timeout = 5.seconds)
 
   // (a) Replicator contains data of all workers (requested and observed status, plus a name index)
-  private val RequestedStatusDataKey: LWWMapKey[WorkerKey, Status] =
-    LWWMapKey[WorkerKey, Status]("projection-registry-requested-status")
+  private val WorkerRequestedStatusDataKey: LWWMapKey[WorkerKey, Status] =
+    LWWMapKey[WorkerKey, Status]("worker-level-requested-status")
+  private val ProjectionRequestedStatusDataKey: LWWMapKey[ProjectionName, Status] =
+    LWWMapKey[ProjectionName, Status]("projection-level-requested-status")
   private val ObservedStatusDataKey: LWWMapKey[WorkerKey, Status] =
     LWWMapKey[WorkerKey, Status]("projection-registry-observed-status")
   private val NameIndexDataKey: LWWMapKey[WorkerKey, WorkerCoordinates] =
     LWWMapKey[WorkerKey, WorkerCoordinates]("projection-registry-name-index")
 
-  replicator ! Subscribe(RequestedStatusDataKey, self)
+  replicator ! Subscribe(WorkerRequestedStatusDataKey, self)
+  replicator ! Subscribe(ProjectionRequestedStatusDataKey, self)
   replicator ! Subscribe(ObservedStatusDataKey, self)
   replicator ! Subscribe(NameIndexDataKey, self)
 
   // (b) Keep a local copy to simplify the implementation of some ops
-  var requestedStatusLocalCopy: Map[WorkerKey, Status]      = Map.empty[WorkerKey, Status]
-  var observedStatusLocalCopy: Map[WorkerKey, Status]       = Map.empty[WorkerKey, Status]
+  var workerRequestedStatusLocalCopy: Map[WorkerKey, Status]     = Map.empty[WorkerKey, Status]
+  var projectionRequestedStatusLocalCopy: Map[WorkerKey, Status] = Map.empty[ProjectionName, Status]
+  var observedStatusLocalCopy: Map[WorkerKey, Status]            = Map.empty[WorkerKey, Status]
+  // the nameIndexLocalCopy is special: it contains consistent data about the local worker and it eventually
+  // becomes a copy of the replicated data. This special treatment prevents a race condition when reading from it.
+  // This is safe because the name index CRDT is monotonic and each local copy just will be equal or a bit ahead
+  // of the CRDT (both eventually being identical)
   var nameIndexLocalCopy: Map[WorkerKey, WorkerCoordinates] = Map.empty[WorkerKey, WorkerCoordinates]
```

Comment:
While testing the actor I noticed `nameIndexLocalCopy` is particular and requires a different updating/copying mechanism than the rest of `xyzLocalCopy`. The advantage is that write ops are started on the worker-side (not the user-side) and, more importantly, it's a monotonic map with a known limit of values so it's very reason to reason about it and merge the `nameIndexLocalCopy` with the `NameIndexDataKey` CRDT.